src/panels/KaitoModelsPanel.ts

```diff
@@ -96,21 +100,102 @@ export class KaitoModelsPanelDataProvider implements PanelDataProvider<"kaitoMod
     nullIsFalse(value: boolean | null): boolean {
         return value ?? false;
     }
+    parseGPU(gpuRequirement: string): [string, number] {
+        const regex = /^Standard_NC(\d+)(ads_A100_v4|s_v3)$/;
+        const match = gpuRequirement.match(regex);
+        if (match) {
+            const cpus = parseInt(match[1], 10);
+            let family: string;
+            switch (match[2]) {
+                case "ads_A100_v4":
+                    family = "StandardNCADSA100v4Family";
+                    break;
+                case "s_v3":
+                    family = "standardNCSv3Family";
+                    break;
+                default:
+                    throw new Error("Unknown gpu format");
+            }
+
+            return [family, cpus];
+        } else {
+            throw new Error("Unknown gpu format");
+        }
+    }
 
-    private async handleDeployKaitoRequest(model: string, yaml: string, webview: MessageSink<ToWebViewMsgDef>) {
+    private async handleDeployKaitoRequest(
+        model: string,
+        yaml: string,
+        gpu: string,
+        webview: MessageSink<ToWebViewMsgDef>,
+    ) {
         this.cancelToken = false;
+        if (this.checkingGPU) {
+            return;
+        }
         try {
+            this.checkingGPU = true;
+            let quotaAvailable = false;
+            let getResult = null;
+            await longRunning(`Checking if workspace exists...`, async () => {
+                const getCommand = `get workspace workspace-${model}`;
+                getResult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, getCommand);
+            });
+            if (getResult === null || failed(getResult)) {
+                await longRunning(`Verifying available subscription quota for deployment...`, async () => {
+                    const [gpuFamily, requiredCPUs] = this.parseGPU(gpu);
+                    void requiredCPUs;
+                    const computeClient = getComputeManagementClient(this.sessionProvider, this.subscriptionId);
+                    const containerServiceClient = getAksClient(this.sessionProvider, this.subscriptionId);
+                    const cluster = await containerServiceClient.managedClusters.get(
+                        this.resourceGroupName,
+                        this.clusterName,
+                    );
+                    const quotas = computeClient.usageOperations.list(cluster.location);
+                    let foundQuota = null;
+                    for await (const quota of quotas) {
+                        if (quota.name.value === gpuFamily) {
+                            foundQuota = quota;
+                            break;
+                        }
+                    }
+                    if (!foundQuota || !(foundQuota.currentValue + requiredCPUs <= foundQuota.limit)) {
```

Comment:
💡 lets extract this from one big function to something like this? What do you think?

- Helper Functions: `findMatchingQuota`, `isQuotaSufficient`, and `promptForQuotaIncrease` have been extracted for clarity.
- Error Handling: The `promptForQuotaIncrease` function encapsulates the error message display and handling of user interaction.

```*.ts
if (getResult === null || failed(getResult)) {
    await longRunning(`Verifying available subscription quota for deployment...`, async () => {
        const [gpuFamily, requiredCPUs] = this.parseGPU(gpu);
        const computeClient = getComputeManagementClient(this.sessionProvider, this.subscriptionId);
        const containerServiceClient = getAksClient(this.sessionProvider, this.subscriptionId);
        const cluster = await containerServiceClient.managedClusters.get(
            this.resourceGroupName,
            this.clusterName,
        );
        const quotas = computeClient.usageOperations.list(cluster.location);
        
        const foundQuota = await findMatchingQuota(quotas, gpuFamily);
        
        if (!foundQuota || !isQuotaSufficient(foundQuota, requiredCPUs)) {
            await promptForQuotaIncrease();
        } else {
            quotaAvailable = true;
        }
    });
} else {
    quotaAvailable = true;
}

// Helper functions
async function findMatchingQuota(quotas, gpuFamily) {
    for await (const quota of quotas) {
        if (quota.name.value === gpuFamily) {
            return quota;
        }
    }
    return null;
}

function isQuotaSufficient(quota, requiredCPUs) {
    return quota.currentValue + requiredCPUs <= quota.limit;
}

async function promptForQuotaIncrease() {
    const selection = await vscode.window.showErrorMessage(
        `Your current Azure subscription doesn't have enough quota to deploy this model. Proceed to request a quota increase.`,
        "Learn More"
    );

    if (selection === "Learn More") {
        vscode.env.openExternal(
            vscode.Uri.parse("https://learn.microsoft.com/en-us/azure/quotas/quickstart-increase-quota-portal")
        );
    }
}

```